gtest_parallel.py

```diff
@@ -341,18 +341,15 @@ def __construct_from_failure(self, task):
     it may not have generated valid XML. Handle this case by manually-constructing
     conformant XML from stdout/stderr output
     """
-    try:
-      return ET.parse(task.xml_file)
```

Comment:
Would it be possible to inspect the xml if the parse doesn't throw and fall back if there is no error?

Reply:
It's possible, but there's a side effect to doing it this way that works in our favor. I've noticed that this method concatenates all failure/error messages into a single `<failure>` element for each test, which improves error reporting through Jenkins. 

For each `EXPECT` failure, `googletest` will append a new `<failure>` element to the parent `<testcase>` element, but Jenkins (at least _our_ Jenkins) can't process more than one `<failure>` per test. Concatenating all of the failures together means that we can see all of them in the Jenkins/vTest dashboard, instead of just the first one that was reported.